### ISSUE TITLE
[Tests] Disable that horrible `test_wrapper_fail_obj_large()` unit test that everybody justifiably hates

### ISSUE THREAD

**Issue by WeepingClown13** [beartype/beartype#362]
Hi, first of all, thanks for maintaining beartype, we have been making great use of it at work.

I currently have packaged beartype for debian and it is available in debian unstable/testing (`sudo apt install python3-beartype`). I noticed that one of the tests (`beartype_test/a00_unit/a70_decor/test_decorwrapper.p::ytest_wrapper_fail_obj_large`) fails consistently by timing out  on armhf architecture(the limit is 15 seconds, it takes from barely over 15 to 17+ seconds). I tried several test runs for the same in a local armhf schroot and it keeps the same behaviour. Considering how slow the whole thing seems in the arch, I suppose this is just a sad but inevitable outcome in all probability.

I'd like for the tests to pass, so I am adding the following patch to get it to work.

```bash
diff --git a/beartype_test/a00_unit/a70_decor/test_decorwrapper.py b/beartype_test/a00_unit/a70_decor/test_decorwrapper.py
index 6827233..a0cbe22 100644
--- a/beartype_test/a00_unit/a70_decor/test_decorwrapper.py
+++ b/beartype_test/a00_unit/a70_decor/test_decorwrapper.py
@@ -57,8 +57,16 @@ def test_wrapper_fail_obj_large() -> None:
     # Number of seconds after which the call to the wrapper function defined
     # below passed the worst-called structure defined below is automatically
     # failed. This effectively acts as a poor man's timeout. We are that man.
+
     TIME_MAX = 15.0
 
+    # But armhf arch takes the race a bit more nice and easy and misses the
+    # limit, albeit barely
+    import platform
+
+    if 'arm' in platform.machine().lower() and platform.architecture()[0] == '32bit':
+        TIME_MAX = 20.0
+
     # Arbitrary callable accepting a deeply nested data structure.
     @beartype
     def behold(the_great_destroyer_of_apps: List[List[List[int]]]) -> int:
```

Do you think the same can be added here? I would be happy to open a PR for the same if so.  Or if there are better solutions, I'd be happy to hear them.

**Comment by leycec** [comment]
Thanks so much for the kind words, @WeepingClown13. What a username... *and what a moé avatar!* **Gah!** :heart_eyes_cat: 

You're right about everything, of course. The `test_wrapper_fail_obj_large()` unit test is hardware- and computational load-sensitive, which is horrible. That test exists mostly to avoid performance regressions in exception handling – but I never thought of a particularly good way to make it scalable across divergent hardware and load scenarios. In short, that test sucks.

It's probably best if I conditionally disable that test for everyone except me. I'd still like to locally run that test, because I'm me and just can't give up on a sucky test. Inspired by your patch, I'll end up doing something like:

```python
def test_wrapper_fail_obj_large() -> None:
    from beartype._util.os.utilosshell import get_shell_var_value_or_none
    if get_shell_var_value_or_none('USER') != 'leycec':
       return
    ...
```

Thus did a sucky test become kinda even suckier. :laughing:

**Comment by WeepingClown13** [comment]
@leycec haha, your call :P
I currently packaged and uploaded v.0.18.2 to debian with my patch, and I am happy as long as everything works. Please make any unbearable change you want, hopefully one that allows me to carry on later without having to maintain such a patch downstream and doesn't break anyone's [workflow](https://xkcd.com/1172/).

PS: other leycecs on armhf be roaring now!

**Comment by leycec** [comment]
> hopefully one that allows me to carry on later without having to maintain such a patch downstream and doesn't break anyone's [workflow](https://xkcd.com/1172/).

**Absolutely.** That's the priority. Preserve backward compatibility. Preserve @leycec's sanity. Preserve your workflow. Avoid pitchforks, flaming torches, upside-down crosses, and garlic. :vampire_man: 

I've noticed that other competing distros we don't talk about here (*...okay, it's Arch and Gentoo*) just unconditionally ignore this test in their package builds for @beartype. That's pretty awful and a clear indication that this test needs to go for everybody who is not me.

Oh – and thanks so much for the hard packaging work! I personally know how brutal packaging can get, because that is something I once did when I was young, idealistic, and had hair. You're doing the Good Work™. Bless you.

**Comment by leycec** [comment]
**Resolved by 5bdbe240a84cb.**

![](https://media1.tenor.com/m/0vSCasGK6-oAAAAd/cat-funny.gif)

**Comment by WeepingClown13** [comment]
Awesome, thanks for the quick resolution and all the hardwork. Hope we can keep the communication smooth in the future as well :)

PS: personally a big fan of all the puns and grammar, even more than the library itself xD biggest reason I packaged it for Debian!
